tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitIntegration.cs

```diff
@@ -36,23 +36,25 @@ internal static Scope CreateScope(ITest currentTest, Type targetType)
                 return null;
             }
 
+            Common.Prepare(testMethod);
+
             string testFramework = "NUnit";
             string fullName = currentTest.FullName;
             string composedTestName = currentTest.Name;
 
             string testName = testMethod.Name;
-            string testSuite = testMethod.DeclaringType?.FullName;
-            string testBundle = testMethod.DeclaringType?.Assembly?.GetName().Name;
+            string testSuite = testMethod.DeclaringType?.FullName ?? string.Empty;
+            string testBundle = testMethod.DeclaringType?.Assembly.GetName().Name ?? string.Empty;
 
             // Extract the test suite from the full name to support custom fixture parameters and test declared in base classes.
             if (fullName.EndsWith("." + composedTestName))
             {
                 testSuite = fullName.Substring(0, fullName.Length - (composedTestName.Length + 1));
             }
 
-            string skipReason = null;
+            string? skipReason = null;
 
-            Scope scope = Tracer.Instance.StartActiveInternal("nunit.test");
+            Scope? scope = Tracer.Instance.StartActiveInternal("nunit.test");
```

Comment:
Because the file is now marked as nullable, and later in the same method when we mark the span as skipped we set the scope as null. Meaning that the declaration must be `Scope?`